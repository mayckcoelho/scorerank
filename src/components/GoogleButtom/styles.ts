import styled from 'styled-components/native';

export const StyledButtom = styled.TouchableOpacity`
    margin: 10px 0;
    border: 1px solid #edeef2;
    border-radius: 4px;
    padding: 16px;
    align-items: center;
    justify-content: center;
    background-color: #fff;

    flex-direction: row;
`

export const ButtomText = styled.Text`
    margin-left: 10px;
    font-weight: bold;
    color: #455675;
    font-size: 15px;
`