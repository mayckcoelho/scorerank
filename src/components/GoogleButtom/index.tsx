import React from 'react';

import { StyledButtom, ButtomText } from './styles';
import Icon from "react-native-vector-icons/FontAwesome";
Icon.loadFont();

const GoogleButtom: React.FC = () => (
    <StyledButtom>
        <Icon name="google" size={16} color="#ea4335" />
        <ButtomText>Continuar com Google</ButtomText>
    </StyledButtom>
);

export default GoogleButtom;