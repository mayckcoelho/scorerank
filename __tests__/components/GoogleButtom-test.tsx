import React from 'react';
import GoogleButtom from '../../src/components/GoogleButtom';
import renderer from 'react-test-renderer';

describe('GoogleButtom should render correctly', () => {
    it('match snapshot', () => {
        const googleButtom = renderer
            .create(<GoogleButtom />)
            .toJSON();
        expect(googleButtom).toMatchSnapshot();
    });
})