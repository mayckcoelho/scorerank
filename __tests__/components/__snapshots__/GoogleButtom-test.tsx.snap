// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleButtom should render correctly match snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#edeef2",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 10,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#ea4335",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï† 
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#455675",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginLeft": 10,
        },
      ]
    }
  >
    Continuar com Google
  </Text>
</View>
`;
